### ISSUE TITLE
Autorank 4: New Paths.yml 

### ISSUE THREAD

**Issue by nathan93** [Staartvin/Autorank-2#445]
I am not a fan of this update that replaces the AdvancedConfig with this new Paths.yml, but I guess I'll have to live with it. 

The wiki isn't giving an explanation for what I need. I simply want players to be ranked up if they meet certain requirements, none of this other extra complicated stuff. My AdvancedConfig on Autorank 3.8 looked like this http://pastebin.com/thtVjFTJ

All I want is simply for that to work. How might I do that with this new update?

**Comment by Staartvin** [comment]
I'm sorry that you're having troubles with the new format. I tried to keep the format close to what Autorank originally had. The good news is that Paths.yml and the AdvancedConfig share more or less the same format. The biggest change you'll have to perform is that each path will have a prerequisite that is the 'in group'. 

Take a look at your Member group. You would create a new path, for example Member to Novice, that has a prerequisite 'in group' with value 'Member'. The requirements and results sections can be copied from the AdvancedConfig.  

That's it!

**Comment by nathan93** [comment]
Do you have an example of what the configuration of Paths.yml might look like for Member to Novice? I understand that you need a prerequisite, however what I'm not understanding is how to format it in the configuration.

**Comment by Staartvin** [comment]
Here's an example for the Member to Novice group:

```YAML

Member to Novice:
    prerequisites:
        in group:
            value: Member
    requirements:
        mcmmo power level:
            value: 100
        time:
            value: 2h
    results:
        rank change: Novice
        command: 'bc &p has just been promoted to &bNovice&a, and earned &91 crate key!; crate givekey &p SionCrate 1; tell &p You have just earned 1 crate key!'
        message: 'Congratulations, you are now a Novice!'
```

There you go! Looks a lot like the AdvancedConfig.yml!

**Comment by nathan93** [comment]
From what I understand, that is one path? How do I make it so they players don't have to consistently choose the path they want so they can get the ranks they desire? Or am I misunderstanding the setup?

**Comment by Staartvin** [comment]
That is correct, this is one path. If you don't want players to explicitly choose (via the /ar choose command) what path they want, you can use [the 'auto choose' option](https://github.com/Armarr/Autorank-2/wiki/How-to-configure-the-Paths.yml-file%3F#automatically-assign-a-path-to-a-player). It comes down to adding a _*auto choose: true*_ to the options of a path.

**Comment by nathan93** [comment]
So just create more paths (i.e. Novice to Apprentice, etc.) and in each path set "autochoose" to true?

**Comment by Staartvin** [comment]
Yes, don't forget the space in the 'auto choose'. That's how it works! Note that you need Autorank **v4.0.1** for this to work.

**Comment by nathan93** [comment]
It's still not working how I'd like to. I formatted it like you showed, however it still doesn't "auto rank" me when I meet the requirements set, and when I check the requirements in game, it says the only requirements are to be in a specific group.

**Comment by Staartvin** [comment]
Can you show me your Paths file, Settings file and please do /ar debug and provide that file as well!

**Comment by nathan93** [comment]
[Paths](http://pastebin.com/m4WtDBVA), [Settings](http://pastebin.com/VXNHefke), [Debug](http://pastebin.com/n7wLA4zY)

**Comment by Staartvin** [comment]
What does it show you when you do /ar check when you are in one of the paths?

**Comment by nathan93** [comment]
When I set myself as a member, I still have to manually select the ranking path, when I do /ar check in the path "Member to Novice" it will promote me and run the commands it's supposed to, but then I have to manually select the next path. It looks like auto choose isn't working (I have 4.0.1). I set myself as member, chose path "Member to Novice," ran /ar check, and was promoted to Novice. Then I tried selecting "Novice to Apprentice" and was told I did not meet the requirements to select that path.
